src/lakefs_spec/spec.py

```diff
@@ -424,6 +481,9 @@ def __init__(
             size=size,
             **kwargs,
         )
+        if mode == "wb":
+            repository, branch, resource = parse(path)
+            ensure_branch(self.fs.client, repository, branch, self.fs.source_branch)
```

Comment:
Can we merge this with the warning above the `super()` call? Just move the warning here and delete the first if-branch.